app/controllers/nonsipp/loansmadeoroutstanding/LoansCheckAndUpdateController.scala

```diff
@@ -16,58 +16,71 @@
 
 package controllers.nonsipp.loansmadeoroutstanding
 
+import services.SchemeDateService
 import viewmodels.implicits._
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import com.google.inject.Inject
-import cats.implicits.toShow
+import utils.nonsipp.summary.LoansCheckAnswersUtils
+import utils.IntUtils.toInt
+import utils.IntUtils.given
 import controllers.actions.IdentifyAndRequireData
-import pages.nonsipp.common.OtherRecipientDetailsPage
-import pages.nonsipp.loansmadeoroutstanding._
+import models.NormalMode
+import play.api.i18n.MessagesApi
 import config.RefinedTypes.Max5000
 import controllers.PSRController
-import views.html.ContentTablePageView
+import views.html.PrePopCheckYourAnswersView
 import models.SchemeId.Srn
-import utils.IntUtils.{toInt, toRefined5000}
-import controllers.nonsipp.loansmadeoroutstanding.LoansCheckAndUpdateController._
-import utils.DateTimeUtils.localDateShow
-import models.{IdentitySubject, Money, NormalMode}
-import play.api.i18n.MessagesApi
-import viewmodels.DisplayMessage
-import viewmodels.DisplayMessage.{ListMessage, Message, ParagraphMessage}
-import viewmodels.models.{ContentTablePageViewModel, FormPageViewModel}
-
-import java.time.LocalDate
+import viewmodels.DisplayMessage.Message
+import viewmodels.models.{CheckYourAnswersSection, CheckYourAnswersViewModel, FormPageViewModel}
 
 class LoansCheckAndUpdateController @Inject() (
   override val messagesApi: MessagesApi,
   identifyAndRequireData: IdentifyAndRequireData,
   val controllerComponents: MessagesControllerComponents,
-  view: ContentTablePageView
+  schemeDateService: SchemeDateService,
+  view: PrePopCheckYourAnswersView
 ) extends PSRController {
-
   def onPageLoad(srn: Srn, index: Int): Action[AnyContent] = identifyAndRequireData(srn) { implicit request =>
-    (
-      for {
-        recipientName <- List(
-          request.userAnswers.get(IndividualRecipientNamePage(srn, index)),
-          request.userAnswers.get(CompanyRecipientNamePage(srn, index)),
-          request.userAnswers.get(PartnershipRecipientNamePage(srn, index)),
-          request.userAnswers.get(OtherRecipientDetailsPage(srn, index, IdentitySubject.LoanRecipient)).map(_.name)
-        ).flatten.headOption.getOrRecoverJourney
-        datePeriodDetails <- request.userAnswers.get(DatePeriodLoanPage(srn, index)).getOrRecoverJourney
-        amountOfTheLoan <- request.userAnswers.get(AmountOfTheLoanPage(srn, index)).getOrRecoverJourney
-      } yield Ok(
-        view(
-          viewModel(
-            srn,
-            index,
-            recipientName,
-            datePeriodDetails._1,
-            amountOfTheLoan.loanAmount
+    LoansCheckAnswersUtils(schemeDateService)
+      .summaryData(srn, index, NormalMode)
+      .map { data =>
+        val sections = LoansCheckAnswersUtils(schemeDateService)
+          .viewModel(
```

Comment:
Is it possible to simplify this, like it was done in land or property journey?
https://github.com/hmrc/pension-scheme-return-frontend/compare/6db6ab2e26b1ea69691eb10d918e9a7ec7f7d7ca..33e52073adb202c53b3afd9c6d0ba09a77bf6594